src/libraries/System.Private.CoreLib/src/System/Globalization/StringInfo.cs

```diff
@@ -153,44 +196,19 @@ public static int[] ParseCombiningCharacters(string str)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.str);
             }
 
-            // This method is optimized for small-ish strings.
-            // If a large string is seen we'll go down a slower code path.
-
-            if (str.Length > 256)
-            {
-                return ParseCombiningCharactersForLargeString(str);
-            }
-
-            Span<int> baseOffsets = stackalloc int[str.Length];
-            int graphemeClusterCount = 0;
-
+            ValueListBuilder<int> builder = new ValueListBuilder<int>(stackalloc int[16]); // 16 arbitrarily chosen
```

Comment:
If you think it's worthwhile, we could get away with a larger number here.  We're often using anywhere from 256 to 1024 for bytes, so we could make this 64, for example, and it would still be inline with what we do elsewhere.  Up to you.

Reply:
(Especially since, reading the previous code, it was stackalloc'ing up to 256 ints.)

Reply:
64 it is! :)